aten/src/ATen/native/cuda/ReflectionPad.cu

```diff
@@ -403,26 +625,65 @@ void reflection_pad2d_backward_out_template(
   int64_t size_y = nplane;
   int64_t size_z = nbatch;
 
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16,
-    input.scalar_type(), "reflection_pad2d_backward_out_template", [&] {
-
-      for (int64_t block_y = 0; block_y < size_y; block_y += 65535) {
-        int64_t block_y_size = std::min(size_y - block_y, static_cast<int64_t>(65535));
-        for (int64_t block_z = 0; block_z < size_z; block_z += 65535) {
-          int64_t block_z_size = std::min(size_z - block_z, static_cast<int64_t>(65535));
-
-          dim3 grid_size(at::ceil_div(output_plane_size, static_cast<int64_t>(256)), block_y_size, block_z_size);
-
-          reflection_pad2d_backward_out_kernel<<<
-            grid_size, block_size, 0, at::cuda::getCurrentCUDAStream()>>>(
-              grad_input.mutable_data_ptr<scalar_t>(), grad_output.const_data_ptr<scalar_t>(),
-              input_w, input_h,
-              pad_t, pad_b, pad_l, pad_r, block_y, block_z, nplane);
-          C10_CUDA_KERNEL_LAUNCH_CHECK();
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+      kHalf,
+      kBFloat16,
+      input.scalar_type(),
+      "reflection_pad2d_backward_out_template",
+      [&] {
+        if (at::globalContext().deterministicAlgorithms()) {
+
+          reflection_pad2d_backward_det_out_kernel<<<
+              1024,
+              256,
```

Comment:
I just put these values in for now. They should be variables though and perhaps more thought should go into it.